### PR TITLE
refactor/ Huobi aiohttp websocket

### LINKED ISSUE #4408: Replace websockets library with aiohttp library - Huobi
See epic https://github.com/CoinAlpha/hummingbot/issues/3941 for the rationale of this task.

## Definition of Done
* [ ] Replace `websockets` library with `aiohttp` in `Huobi` connector
* [ ] Connector should be working without issues after the change

┆Issue is synchronized with this [Clickup task](https://app.clickup.com/t/1pd0g0a) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Update Huobi connector to use `WebAssistantsFactory`.
Closes #4408 

**Tests performed by the developer**:
Update `HuobiExchange`, `HuobiAPIOrderBookDataSource`, `HuobiUserStreamDataSource`
Add `HuobiWSPostProcessor` to handle gzip compressed Market Data messages
Fix `WSConnection` to handle `aiohttp.WSMsgType.BINARY` and `aiohttp.WSMsgType.CLOSE` messages.
Update `WebAssistantsFactory` to include pre_processors and post_processors

**Tips for QA testing**:
Ensure Huobi connector works as usual.

### CODE REVIEW

**Review by petioptrv — CHANGES_REQUESTED** [review]
Left a couple of small comments / suggestions in the code

File: hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py [hunk 1]
```diff
@@ -1,39 +1,45 @@
...
 
 
 class HuobiAPIOrderBookDataSource(OrderBookTrackerDataSource):
 
     MESSAGE_TIMEOUT = 30.0
     PING_TIMEOUT = 10.0
+    HEARTBEAT_INTERVAL = 30.0  # seconds
+    ORDER_BOOK_SNAPSHOT_DELAY = 60 * 60  # expressed in seconds
+
+    TRADE_CHANNEL_SUFFIX = "trade.detail"
```
[the remark this answers is not in the document]
Comment (CrimsonJacket) [reply hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py:41]: According to Huobi's [docs](https://huobiapi.github.io/docs/spot/v1/en/#trade-detail), it is a suffix.

>
\# Orderbook Depth Response
{
  "id": "id1",
  "status": "ok",
  "subbed": "market.btcusdt.depth.step0",
  "ts": 1489474081631 //system response time
}
>
\# Trade Detail Response
{
  "id": "id1",
  "status": "ok",
  "subbed": "market.btcusdt.trade.detail",
  "ts": 1489474081631 //system response time
}

File: hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py [hunk 2]
```diff
@@ -43,188 +49,203 @@ def logger(cls) -> HummingbotLogger:
...
+                    data = ws_response.data
+                    if "subbed" in data:
+                        continue
+                    if "ping" in data:
+                        ping_request = WSRequest(payload={
+                            "pong": data["ping"]
+                        })
+                        await ws.send(request=ping_request)
+                    channel = data.get("ch", "")
+                    if self.TRADE_CHANNEL_SUFFIX in channel:
```
[the remark this answers is not in the document]
Comment (CrimsonJacket) [reply hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py:180]: I agree with these changes. I will update em accordingly.

File: hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py [hunk 3]
```diff
@@ -43,188 +49,203 @@ def logger(cls) -> HummingbotLogger:
...
+                        self._message_queue[self.TRADE_CHANNEL_SUFFIX].put_nowait(data)
+                    if self.ORDERBOOK_CHANNEL_SUFFIX in channel:
```
[the remark this answers is not in the document]
Comment (CrimsonJacket) [reply hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py:182]: I agree with these changes. I will update em accordingly.

File: hummingbot/connector/exchange/huobi/huobi_exchange.pyx [hunk 4]
```diff
@@ -254,28 +255,21 @@ cdef class HuobiExchange(ExchangeBase):
...
-                method=method.upper(),
-                url=url,
-                headers=headers,
-                params=params,
-                data=ujson.dumps(data),
-                timeout=self.API_CALL_TIMEOUT
-            )
+        request = RESTRequest(method=RESTMethod[method.upper()],
+                              url=url,
+                              data=ujson.dumps(data) if data else "",
```
[the remark this answers is not in the document]
Comment (CrimsonJacket) [reply hummingbot/connector/exchange/huobi/huobi_exchange.pyx:266]: Good catch 🎣

File: hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (petioptrv) [reply hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py:41]: My mistake! I think it's time for another coffee

**Review by petioptrv — APPROVED** [review]
Changes LGTM!

**Review by rapcmia — APPROVED** [review]
Tested and working ok, approved on behalf of QA team

**Pull request merged**